### PR TITLE
Closes #1078: Added UI for managing phone feature permissions

### LINKED ISSUE #1078: FNX-4780 ⁃ Manage each type of permission
### Camera Permissions
As a user, I want to know when websites want access to the camera on my device and want to control whether or not they can have access.

### Acceptance criteria
If I have not given Fenix permission to access my camera, I want Android to prompt me to grant Fenix permission. 
If I allow Fenix permission to access my camera, when I arrive at a site requesting access to my camera I want Fenix to prompt me to grant permission to the site. 
If I grant permission to the site, I want the site to access my camera and I want to see an indication that the site has access in the site control panel.
If I do not grant permission to the site (either by not allowing or not making an explicit decision) I want Fenix to deny access to my camera and I want to see an indication that the site is blocked from access in the site control panel.
If I deny Fenix permission to access my camera, when I arrive at site requesting access to my camera I want Fenix to deny the website access to my camera and I want to see an indication that the site is blocked by Android from access in the site control panel but I do not want to be interrupted by a prompt. 

### Microphone Permissions
As a user, I want to know when websites want access to the microphone on my device and want to control whether or not they can have access.

### Acceptance criteria
If I have not given Fenix permission to access my microphone, I want Android to prompt me to grant Fenix permission. 
If I allow Fenix permission to access my microphone, when I arrive at a site requesting access to my camera I want Fenix to prompt me to grant permission to the site. 
If I grant permission to the site, I want the site to access my microphone and I want to see an indication that the site has access in the site control panel.
If I do not grant permission to the site (either by not allowing or not making an explicit decision) I want Fenix to deny access to my microphone and I want to see an indication that the site is blocked from access in the site control panel.
If I deny Fenix permission to access my microphone, when I arrive at site requesting access to my microphone I want Fenix to deny the website access to my microphone and I want to see an indication that the site is blocked by Android from access in the site control panel.

### Location Permissions
As a user, I want to know when websites want access to my location and want to control whether or not they can have access.

### Acceptance criteria
If I have not given Fenix permission to access my location, I want Android to prompt me to grant Fenix permission. 
If I allow Fenix permission to access my location, when I arrive at a site requesting access to my location I want Fenix to prompt me to grant permission to the site. 
If I grant permission to the site, I want the site to access my location and I want to see an indication that the site has access in the site control panel.
If I do not grant permission to the site (either by not allowing or not making an explicit decision) I want Fenix to deny access to my location and I want to see an indication that the site is blocked from access in the site control panel.
If I deny Fenix permission to access my location, when I arrive at site requesting access to my location I want Fenix to deny the website access and I want to see an indication that the site is blocked by Android from access in the site control panel.

### Notification Permissions
As a user, I want to know when websites want to send me notifications and control whether or not they send them. 

### Acceptance criteria
If I have not given Fenix permission to send me notifications, I want Android to prompt me to grant Fenix permission. 
If I allow Fenix permission to send me notifications, when I arrive at a site requesting access to send me notifications I want Fenix to prompt me to grant permission to the site. 
If I grant permission to the site, I want the site to send me notifications and I want to see an indication that the site has access in the site control panel.
If I do not grant permission to the site (either by not allowing or not making an explicit decision) I want Fenix to deny access to send me notifications and I want to see an indication that the site is blocked from access in the site control panel.
If I deny Fenix permission to send me notifications, when I arrive at site requesting access to my location I want Fenix to deny the website access and I want to see an indication that the site is blocked by Android from access in the site control panel.

Site Permissions Flow: https://whimsical.co/QXGiB45gjs1iniprdrWdBG

image

### PR DESCRIPTION
**Pull request by Amejia481**
[Video demo](https://drive.google.com/file/d/1Vs2tWAc8my439tzrsJjELAP-pXCKKLhL/view?usp=sharing) 🍕 
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/res/layout/fragment_manage_site_permissions_feature_phone.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+<LinearLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:tools="http://schemas.android.com/tools"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical"
+        android:layout_marginTop="@dimen/radio_button_preference_vertical">
+
+    <RadioGroup android:layout_width="match_parent"
+                android:layout_height="wrap_content">
```
Comment (boek) [inline app/src/main/res/layout/fragment_manage_site_permissions_feature_phone.xml]: One nit: Tab size on some of these files are 8 spaces instead of 4

Comment (Amejia481) [reply app/src/main/res/layout/fragment_manage_site_permissions_feature_phone.xml]: Thanks I will fix it!

**Comment by Amejia481** [comment]
I fixed the indentation issue and added an extra commit   [Closes #1078: Added UI for managing phone feature permissions](https://github.com/mozilla-mobile/fenix/issues/1229) for fixing a related issue.

**Comment by boek** [comment]
Thanks @Amejia481!

File: app/src/main/res/layout/fragment_manage_site_permissions_feature_phone.xml (hunk 1 above)
Comment (Amejia481) [reply app/src/main/res/layout/fragment_manage_site_permissions_feature_phone.xml]: Fixed

**Pull request merged**